ApplicationLibCode/Commands/RicCreateContourMapPolygonTools.cpp

```diff
@@ -191,46 +193,86 @@ std::vector<std::vector<int>> RicCreateContourMapPolygonTools::convertToBinaryIm
 ///
 //--------------------------------------------------------------------------------------------------
-RimPolygon* RicCreateContourMapPolygonTools::createAndAddBoundaryPolygonFromImage( std::vector<std::vector<int>>  image,
-                                                                                   const RigContourMapProjection* contourMapProjection )
+RimPolygon* RicCreateContourMapPolygonTools::createAndAddBoundaryPolygonsFromImage( std::vector<std::vector<int>>  image,
+                                                                                    const RigContourMapProjection* contourMapProjection )
 {
     if ( !contourMapProjection ) return nullptr;
     if ( image.empty() ) return nullptr;
 
-    std::vector<cvf::Vec3d> polygonDomainCoords;
-    auto                    xVertexPositions = contourMapProjection->xVertexPositions();
-    auto                    yVertexPositions = contourMapProjection->yVertexPositions();
-    auto                    origin3d         = contourMapProjection->origin3d();
-    auto                    depth            = contourMapProjection->topDepthBoundingBox();
+    auto xVertexPositions = contourMapProjection->xVertexPositions();
+    auto yVertexPositions = contourMapProjection->yVertexPositions();
+    auto origin3d         = contourMapProjection->origin3d();
+    auto depth            = contourMapProjection->topDepthBoundingBox();
 
-    auto boundaryPoints = RigPolygonTools::boundary( image );
-    for ( auto [i, j] : boundaryPoints )
+    auto hasAnyValue = []( const std::vector<std::vector<int>>& image ) -> bool
     {
-        double xDomain = xVertexPositions.at( i ) + origin3d.x();
-        double yDomain = yVertexPositions.at( j ) + origin3d.y();
+        for ( const auto& row : image )
+        {
+            for ( int val : row )
+            {
+                if ( val != 0 )
+                {
+                    return true;
+                }
+            }
+        }
 
-        polygonDomainCoords.emplace_back( cvf::Vec3d( xDomain, yDomain, depth ) );
-    }
+        return false;
+    };
 
-    // Epsilon used to simplify polygon. Useful range typical value in [5..50]
-    const double defaultEpsilon = 40.0;
-    RigPolygonTools::simplifyPolygon( polygonDomainCoords, defaultEpsilon );
+    auto        polygonCollection = RimTools::polygonCollection();
+    RimPolygon* currentPolygon    = nullptr;
 
-    if ( polygonDomainCoords.size() >= 3 )
+    while ( hasAnyValue( image ) )
     {
-        auto polygonCollection = RimTools::polygonCollection();
+        std::vector<cvf::Vec3d> polygonDomainCoords;
+        auto                    boundaryPoints = RigPolygonTools::boundary( image );
 
-        auto newPolygon = polygonCollection->appendUserDefinedPolygon();
+        // The area threshold defines the minimum number of pixels to create a polygon for
+        const auto   area          = RigPolygonTools::area( boundaryPoints );
+        const double areaThreshold = 10;
```

Comment:
This threshold should probably be user adjustable, not sure where in the ui the value should be

Reply:
Show list of polygon with normalized area when user is asked to set area threshold.